src/realizations/catchment/Bmi_Fortran_Formulation.cpp

```diff
@@ -104,12 +113,17 @@ string Bmi_Fortran_Formulation::get_output_line_for_timestep(int timestep, std::
     // This probably should never happen, but just to be safe ...
     if (output_var_names.empty()) { return ""; }
 
+    // TODO: add codes to handle both scalar and vector outputs
+    // Convert int to long
+    time_t t_delta = timestep;
     // Do the first separately, without the leading comma
-    *output_text_stream << get_var_value_as_double(output_var_names[0]);
+    //*output_text_stream << "," << get_var_value_as_double(output_var_names[0]);
+    *output_text_stream << get_var_vec_as_double(t_delta, output_var_names[0])[0];
```

Comment:
Comments on [Bmi_Cpp_Formulation.cpp] apply to all `get_output_line_for_timestep` implementations... unifying the code in these would also be beneficial (and in scope since they all have to be touched).... consider, is there a way to make this a superclass method or share some code between these?

Reply:
Just noticed this comment before I wrote mine. Agree.